### ISSUE TITLE
Remove the work around for noUnusedVariable

### ISSUE THREAD

**Issue by egamma** [angelozerr/tslint-language-service#47]
@angelozerr the implementation is currently implementing a workaround for an issue with the tslint unusedVariable rule (https://github.com/angelozerr/tslint-language-service/blob/master/src/index.ts#L171). 
This issue has been addressed and the workaround should be deleted and a new version of the node module published.

**Comment by angelozerr** [comment]
@egamma  do we keep this code for old version of tslint?

**Comment by egamma** [comment]
@angelozerr good point. We can keep the code but change the default behavior so that `disableNoUnusedVariableRule` is off by default. Another solution to make the default tslint version specific.

The setting is mentioned in the docs and this should also be updated.

**Comment by victornoel** [comment]
I wouldn't disable it so soon, because I still see some problems with this rule.

It's not exactly the same as before, but as soon as I enable it, vscode becomes unable to derive the types of elements in complex code (everything becomes any).

Anybody noticed the same? (I'm using tslint-language-service configured in the tsconfig.json, without the vnext tslint extension and everything is working as expected except for this).

**Comment by egamma** [comment]
@victornoel what version of tslint are you using?

**Comment by csvn** [comment]
@egamma I've had some issues that sounds similar to what @victornoel said. But mostly, sometimes when I open files it seems like TS can no longer find modules. Even though the modules are installed and the code compiles without errors.

![image](https://user-images.githubusercontent.com/8770194/30969237-141b58d8-a462-11e7-901b-b94469ceb690.png)

If I remove `"disableNoUnusedVariableRule": false`, then the issue above no longer occurs.

tslint: 5.7.0
tslint-lanugage-service: 0.9.6
typescript: 2.3.4

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "baseUrl": "src",
    "strict": true,
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2016",
      "dom"
    ],
    "plugins": [
      {
        "name": "tslint-language-service",
        "alwaysShowRuleFailuresAsWarnings": true,
        "disableNoUnusedVariableRule": false
      }
    ]
  }
}
```

**Comment by victornoel** [comment]
@egamma tslint 5.7.0, ts 2.5.2.

@csvn yep, I also had this one, but the problem was coming from angular-language-service apparently… ! but maybe it was a mix of the two? ^^

**Comment by victornoel** [comment]
(I tested latest ts 2.5.3 too, it is the same)

**Comment by victornoel** [comment]
and of course there is zero error log that can explain where lies the problem, if you know how I can investigate, I will do it!

**Comment by csvn** [comment]
@victornoel I disabled all Angular VS code extensions, deleted `@angular/language-service` from `node_modules`, and restarted chrome. This did not help with the "Cannot find module ..." error. Can you elaborate about the issue coming from `@angular/language-service`? I can't find anything related to Angular with this.

**Comment by victornoel** [comment]
@csvn it is tricky to really know… and actually, now that you ask, maybe I'm confusing things and it was caused by `disableNoUnusedVariableRule`. In any case, if disabling the rule (aka setting the option to false), fixes the problem, I wouldn't try to look for more.

I don't recommend removing `@angular/language-service` at all, even though anyway it is really used if you have the vscode angular language service loaded I think… (but this extension doesn't always work well in complex projects…).

Anyway, to get back to the question of `disableNoUnusedVariableRule`, there is something fishy there and when I set it to `false`, vscode really gets crazy with types in some cases.
I would like to debug it but I couldn't find how ^^
